python/packages/kagent-adk/tests/unittests/models/test_tls_e2e.py

```diff
@@ -0,0 +1,422 @@
+        )
+
+        # Attempt to connect should fail with SSL verification error
+        async with httpx.AsyncClient(verify=ssl_context) as client:
+            with pytest.raises(httpx.ConnectError):
+                await client.get("https://localhost:8444/health")
+
+
+@pytest.mark.asyncio
+async def test_e2e_with_verification_disabled():
+    """E2E test: Connect successfully with verification disabled.
+
+    This test verifies:
+    1. Start test HTTPS server with self-signed certificate
+    2. Create ModelConfig with verifyDisabled: true (no Secret)
+    3. Create httpx client with verification disabled
+    4. Agent connects successfully despite untrusted certificate
+    5. Warning logs are present in output
+    """
+    with TestHTTPSServer(port=8445):
+        # Create SSL context with verification disabled
+        ssl_context = create_ssl_context(
+            verify_disabled=True,
+            ca_cert_path=None,
+            use_system_cas=True,
+        )
+
+        # Verify that False is returned (httpx special value)
+        assert ssl_context is False
+
+        # Create httpx client with verification disabled
+        async with httpx.AsyncClient(verify=False) as client:
+            # Make request to test server - should succeed despite untrusted cert
+            response = await client.get("https://localhost:8445/health")
+
+            # Verify response
+            assert response.status_code == 200
+            assert response.text == "OK"
+
+
+@pytest.mark.asyncio
+async def test_e2e_with_verification_disabled_logs_warning(caplog):
+    """E2E test: Verify warning logs when verification is disabled."""
+    with caplog.at_level(logging.WARNING):
+        ssl_context = create_ssl_context(
+            verify_disabled=True,
+            ca_cert_path=None,
+            use_system_cas=True,
+        )
+
+        # Verify warning was logged
+        assert "SSL VERIFICATION DISABLED" in caplog.text
+        assert "development/testing" in caplog.text.lower()
+
+
+@pytest.mark.asyncio
+async def test_e2e_with_system_and_custom_ca():
+    """E2E test: Connect with both system CAs and custom CA.
+
+    This test verifies the additive behavior where both system CAs
+    and custom CA are trusted.
+    """
+    with TestHTTPSServer(port=8446):
+        # Create SSL context with both system and custom CAs
+        ssl_context = create_ssl_context(
+            verify_disabled=False,
+            ca_cert_path=str(CA_CERT),
+            use_system_cas=True,
+        )
+
+        # Create httpx client
+        async with httpx.AsyncClient(verify=ssl_context) as client:
+            # Make request to test server - should succeed with custom CA
+            response = await client.get("https://localhost:8446/health")
+            assert response.status_code == 200
+            assert response.text == "OK"
+
+
+@pytest.mark.asyncio
+async def test_e2e_openai_client_with_custom_ca():
```

Comment:
can we add a `test_e2e_openai_client_fails_without_custom_ca` to verify a connection error with the client?